src/Nri/Ui/Modal/V5.elm

```diff
@@ -114,49 +115,77 @@ update config msg model =
     Modal.update config msg model
 
 
+{-| -}
+close : Msg
+close =
+    Modal.close
+
+
+{-| Pass the id of the element that focus should return to when the modal closes.
+-}
+open : String -> Msg
+open =
+    Modal.open
+
+
 {-| -}
 type alias FocusableElementAttrs msg =
-    { onlyFocusableElement : List (Root.Attribute msg)
-    , firstFocusableElement : List (Root.Attribute msg)
-    , lastFocusableElement : List (Root.Attribute msg)
+    { onlyFocusableElement : List (Attribute msg)
+    , firstFocusableElement : List (Attribute msg)
+    , lastFocusableElement : List (Attribute msg)
     }
 
 
 {-| -}
 info :
-    { title : { visibleTitle : Bool, title : String }
+    { visibleTitle : Bool
+    , title : String
     , content : FocusableElementAttrs msg -> Html msg
     , wrapMsg : Msg -> msg
     }
     -> Model
     -> Html msg
 info config model =
-    Modal.view
-        { overlayColor = toOverlayColor Colors.navy
-        , wrapMsg = config.wrapMsg
-        , modalAttributes = modalStyles
-        , title = viewTitle Colors.navy config.title
-        , content = config.content >> toUnstyled
-        }
-        model
-        |> fromUnstyled
+    view { overlayColor = Colors.navy, titleColor = Colors.navy } config model
 
 
 {-| -}
 warning :
-    { title : { visibleTitle : Bool, title : String }
+    { visibleTitle : Bool
+    , title : String
     , content : FocusableElementAttrs msg -> Html msg
     , wrapMsg : Msg -> msg
     }
     -> Model
     -> Html msg
 warning config model =
+    view { overlayColor = Colors.gray20, titleColor = Colors.red } config model
+
+
+view :
+    { overlayColor : Css.Color, titleColor : Css.Color }
+    ->
+        { visibleTitle : Bool
+        , title : String
+        , content : FocusableElementAttrs msg -> Html msg
+        , wrapMsg : Msg -> msg
+        }
+    -> Model
+    -> Html msg
+view { overlayColor, titleColor } config model =
     Modal.view
-        { overlayColor = toOverlayColor Colors.gray20
+        { overlayColor = toOverlayColor overlayColor
         , wrapMsg = config.wrapMsg
         , modalAttributes = modalStyles
-        , title = viewTitle Colors.red config.title
-        , content = config.content >> toUnstyled
+        , title = viewTitle titleColor { title = config.title, visibleTitle = config.visibleTitle }
+        , content =
+            \{ onlyFocusableElement, firstFocusableElement, lastFocusableElement } ->
+                { onlyFocusableElement = List.map Html.Styled.Attributes.fromUnstyled onlyFocusableElement
+                , firstFocusableElement = List.map Html.Styled.Attributes.fromUnstyled firstFocusableElement
+                , lastFocusableElement = List.map Html.Styled.Attributes.fromUnstyled lastFocusableElement
+                }
+                    |> config.content
+                    |> toUnstyled
```

Comment:
👍 yeah, kinda unfortunately that it's so verbose to convert between Html.Styled vs Html, but thanks for doing this!